skt/executable.py

```diff
@@ -328,17 +328,17 @@ def cmd_build(cfg):
         logging.error(exc)
         save_state(cfg, {'buildlog': builder.buildlog})
         retcode = 1
-        return
     except Exception:
         (exc, exc_type, trace) = sys.exc_info()
         raise exc, exc_type, trace
 
-    if cfg.get('buildhead'):
-        ttgz = "%s.tar.gz" % cfg.get('buildhead')
-    else:
-        ttgz = addtstamp(tgz, tstamp)
-    shutil.move(tgz, ttgz)
-    logging.info("tarball path: %s", ttgz)
+    if tgz:
+        if cfg.get('buildhead'):
+            ttgz = "%s.tar.gz" % cfg.get('buildhead')
+        else:
+            ttgz = addtstamp(tgz, tstamp)
+        shutil.move(tgz, ttgz)
+        logging.info("tarball path: %s", ttgz)
```

Comment:
The changelog is a little dry.  Is this shift for tgz to allow the failure path to receive the same comments as the success path?  In other words, what is the goal of this patch?

Reply:
Yes. For both success and failure, we want to include the kernel config and similar report message. So for adding the full stage reporting, we can either duplicate a bunch of the code or just check if the kernel was built and deal with that part in the if clause, if that makes sense.

Reply:
No, it makes sense, and the code seems to point to that.  Just remember to add that to the changelog next time.